### PR TITLE
ci: Update per Q3 audit findings

### PR DESCRIPTION
**Pull request by mishomihov00**
**Description**:

Update per Q3 audit findings

**Related issue(s)**:

Related to #872 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by quiet-node — DISMISSED** [review]
LGTM. 1 qq

File: .github/workflows/foundry_test.yml [hunk 1]
```diff
@@ -20,16 +20,16 @@ jobs:
...
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
         with:
           submodules: recursive
 
       - name: Use Node.js [18]
         uses: actions/setup-node@39370e3970a6d050c480ffad4ff0ed4d3fdee5af # v4.1.0
         with:
           node-version: 18
-          cache: npm
+          #cache: npm
```
[the remark this answers is not in the document]
Comment (mishomihov00) [reply .github/workflows/foundry_test.yml]: @quiet-node Added explanation.

File: .github/workflows/foundry_test.yml [hunk 2]
```diff
@@ -20,16 +20,16 @@ jobs:
...
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
         with:
           submodules: recursive
 
       - name: Use Node.js [18]
         uses: actions/setup-node@39370e3970a6d050c480ffad4ff0ed4d3fdee5af # v4.1.0
         with:
           node-version: 18
-          cache: npm
+          #cache: npm Disabling this because it causes the workflow to hang and eventually timeout
```
Comment (quiet-node) [inline .github/workflows/foundry_test.yml:32]: Just curious why don't we remove it instead of commenting it out? Perhaps we might use it later?

**Review by quiet-node — DISMISSED** [review]

File: .github/workflows/foundry_test.yml (hunk 2 above)
Comment (mishomihov00) [reply .github/workflows/foundry_test.yml:32]: Yeah, I want to be clearly visible what was enabled before and why it's disabled. That will prevent someone adding it with good intension, but involuntarily breaking the checks.

**Pull request closed without merging**